apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/index.jsx

```diff
@@ -14,6 +15,8 @@ import EnableLearnPageStagingUrlToggle from "./EnableLearnPageStagingUrlToggle";
 
 const SectionDeveloper = () => {
   const { t } = useTranslation();
+  const location = useLocation();
+  const shouldOpenFeatureFlags = !!(location.state && location.state.shouldOpenFeatureFlags);
```

Comment:
As this logic is specific to the `FeatureFlagsSettings` component and not reused anywhere in this component, it could be moved there (with a `useEffect(() => {...}, []}` that initialises the state), what do you think?

Also you can simplify this code: `!!location.state?.shouldOpenFeatureFlags` or `Boolean(location.state?.shouldOpenFeatureFlags)`

Reply:
I am not sure we can use `useLocation` inside a `useEffect`

Reply:
You can't, you call useLocation, then use the result in the useEffect